sklearn/preprocessing/_function_transformer.py

```diff
@@ -237,7 +224,16 @@ def fit(self, X, y=None):
         self : object
             FunctionTransformer class instance.
         """
-        X = self._check_input(X, reset=True)
+        X = validate_data(
+            self,
+            X,
+            reset=True,
+            accept_sparse=self.accept_sparse,
+            dtype="numeric",
+            ensure_all_finite="allow-nan",
```

Comment:
Why did you set it to ``allow-nan`` ? The default is ``None`` which corresponds to ``True``. Note that if validate=False, check_array is skipped so ensure_all_finite is irrelevant.